test/text.c

```diff
@@ -0,0 +1,358 @@
+        EXPECT_GT(GetStringWidth(fontId, gMovesInfo[move].name, 0), widthPx);
+        break;
+    default:
+        EXPECT_LE(GetStringWidth(fontId, gMovesInfo[move].name, 0), widthPx);
+        break;
+    }
+}
+
+TEST("Move names fit on TMs & HMs Bag Screen")
+{
+    u32 i;
+    const u32 fontId = FONT_NARROWER, widthPx = 7 + 7 * 8;
+    u32 move = MOVE_NONE;
+    for (i = 1; i < ITEMS_COUNT; i++)
+    {
+        if (gItemsInfo[i].pocket == POCKET_TM_HM)
+        {
+            PARAMETRIZE_LABEL("%S", gMovesInfo[gItemsInfo[i].secondaryId].name) { move = gItemsInfo[i].secondaryId; }
+        }
+    }
+    EXPECT_LE(GetStringWidth(fontId, gMovesInfo[move].name, 0), widthPx);
+}
+
+TEST("Move names fit on Move Relearner Screen")
+{
+    u32 i;
+    const u32 fontId = FONT_NARROWER, widthPx = 9 * 8;
+    u32 move = MOVE_NONE;
+    for (i = 1; i < MOVES_COUNT; i++)
+    {
+        PARAMETRIZE_LABEL("%S", gMovesInfo[i].name) { move = i; }
+    }
+    EXPECT_LE(GetStringWidth(fontId, gMovesInfo[move].name, 0), widthPx);
+}
+
+TEST("Item names fit on Bag Screen (list)")
+{
+    u32 i;
+    const u32 fontId = FONT_NARROWER;
+    const u32 tmHmBerryWidthPx = 7 + 8 * 8, restWidthPx = 11 * 8;
+    u32 item = ITEM_NONE;
+    for (i = 1; i < ITEMS_COUNT; i++)
+    {
+        PARAMETRIZE_LABEL("%S", gItemsInfo[i].name) { item = i; }
+    }
+    if (gItemsInfo[item].pocket == POCKET_TM_HM || gItemsInfo[item].pocket == POCKET_BERRIES)
+        EXPECT_LE(GetStringWidth(fontId, gItemsInfo[item].name, 0), tmHmBerryWidthPx);
+    else
+        EXPECT_LE(GetStringWidth(fontId, gItemsInfo[item].name, 0), restWidthPx);
+}
+
+TEST("Item plural names fit on Bag Screen (left box)")
+{
+    u32 i;
+    // -6 for the question mark in FONT_NORMAL.
+    const u32 fontId = FONT_NARROWER, widthPx = 102 - 6;
+    u32 item = ITEM_NONE;
+    u8 pluralName[ITEM_NAME_PLURAL_LENGTH + 1];
+    for (i = 1; i < ITEMS_COUNT; i++)
+    {
+        PARAMETRIZE_LABEL("%S", gItemsInfo[i].name) { item = i; }
+    }
+    CopyItemNameHandlePlural(item, pluralName, 2);
+    EXPECT_LE(GetStringWidth(fontId, pluralName, 0), widthPx);
+}
+
+TEST("Item plural names fit on PC storage (left box)")
+{
+    u32 i;
+    // -6 for the question mark in FONT_NORMAL.
+    const u32 fontId = FONT_NARROWER, widthPx = 13 * 8 - 6;
+    u32 item = ITEM_NONE;
+    u8 pluralName[ITEM_NAME_PLURAL_LENGTH + 1];
+    for (i = 1; i < ITEMS_COUNT; i++)
+    {
+        PARAMETRIZE_LABEL("%S", gItemsInfo[i].name) { item = i; }
+    }
+    CopyItemNameHandlePlural(item, pluralName, 2);
+    EXPECT_LE(GetStringWidth(fontId, pluralName, 0), widthPx);
+}
+
+TEST("Item names fit on Pokemon Storage System")
+{
+    u32 i;
+    const u32 fontId = FONT_SMALL_NARROWER, widthPx = 7 * 8 - 6;
+    u32 item = ITEM_NONE;
+    for (i = 1; i < ITEMS_COUNT; i++)
+    {
+        if (gItemsInfo[i].importance) continue;
+        PARAMETRIZE_LABEL("%S", gItemsInfo[i].name) { item = i; }
+    }
+    switch (item)
```

Comment:
Can't this be a check based on item name length so it supports custom item names?

Reply:
That wouldn't make sense—the point here is to list just the items that are too long.

Reply:
After a talk on Discord, I now get what you meant by this ^^;